tests/main/views/test_notifications.py

```diff
@@ -53,3 +53,21 @@ def test_unsubscribe_content(self):
         res = self.client.get("/user/notifications/user-research")
         assert res.status_code == 200
         assert "Unsubscribe from the user research mailing list" in res.get_data(as_text=True)
+
+    @mock.patch('app.main.views.notifications.data_api_client', autospec=True)
+    def test_user_research_opt_in(self, data_api_client):
+        self.login_as_buyer()
+        self.client.post("/user/notifications/user-research", data={"user_research_opt_in": "True"})
+        self.assert_flashes('Your preference has been saved', expected_category='success')
+        assert data_api_client.update_user.call_args_list == [
+            mock.call(123, updater='buyer@email.com', user_research_opted_in=True)
+        ]
+
+    @mock.patch('app.main.views.notifications.data_api_client', autospec=True)
+    def test_user_research_opt_out(self, data_api_client):
+        self.login_as_buyer()
+        self.client.post("/user/notifications/user-research", data={})
+        self.assert_flashes('Your preference has been saved', expected_category='success')
```

Comment:
@TheDoubleK - it now redirects to the dashboards, as it's going to a different app I've had to assert the flash messages using this function. Is this the correct way?

Reply:
This looks fine, however we're in the process of revamping the flash messages across all apps so you might want to look at the latest master of the Supplier FE to check how it will be displayed.

Reply:
(Ah - it's not on master yet! See https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/845)